yapapi/agreements_pool.py

```diff
@@ -38,9 +38,15 @@ class BufferedAgreement:
 class AgreementsPool:
     """Manages proposals and agreements pool"""
 
-    def __init__(self, job_id: str, emitter: Callable[[events.Event], None]):
+    def __init__(
+        self,
+        job_id: str,
+        emitter: Callable[[events.Event], None],
+        offer_recycler: Callable[[OfferProposal], None],
```

Comment:
```suggestion
        offer_recycler: Callable[[OfferProposal], None] = lambda _offer: None,
```

Reply:
As discussed, I'm strongly against adding defaults that 
* have values that are questionable from the logic POV (why would we want to not-recycle such offers?)
* are always overwritten

even if this simplifies the tests a little.